libs/core/src/lib/popover/cdk-overlay/cdk-popover.component.ts

```diff
@@ -0,0 +1,395 @@
+        private _overlay: Overlay,
+        private _viewportRuler: ViewportRuler,
+        @Optional() private _rtlService: RtlService
+    ) {
+        super();
+
+        /** Merge observables - close or destroy */
+        this._refresh$ = merge(this.isOpenChange, this._onDestroy$);
+    }
+
+    /** @hidden */
+    ngOnInit(): void {
+        if (!this.scrollStrategy) {
+            this.scrollStrategy = this._overlay.scrollStrategies.reposition();
+        }
+
+    }
+
+    /** @hidden */
+    ngAfterViewInit(): void {
+        this._initialised = true;
+        this._refreshTriggerListeners();
+        if (this.isOpen) {
+            this.open();
+        }
+    }
+
+    /** @hidden */
+    ngOnChanges(changes: SimpleChanges): void {
+        if (!this._initialised) {
+            return;
+        }
+
+        if (changes['isOpen']) {
+            if (changes['isOpen'].currentValue) {
+                this.open();
+            } else {
+                this.close();
+            }
+        }
+
+        if (changes['triggers']) {
+            this._refreshTriggerListeners();
+        }
+    }
+
+    /** @hidden */
+    ngOnDestroy(): void {
+        this._onDestroy$.next();
+        this._onDestroy$.complete();
+        if (this._overlayRef) {
+            this._overlayRef.detach();
+        }
+    }
+
+    /** Toggles the popover open state */
+    toggle(): void {
+        if (this.isOpen) {
+            this.close();
+        } else {
+            this.open();
+        }
+    }
+
+    /** Closes the popover. */
+    close(): void {
+        if (this._overlayRef) {
+            this._removeArrowStyles();
+            this._overlayRef.dispose();
+            this._changeDetectorReference.detectChanges();
+            if (this.isOpen) {
+                this.isOpenChange.emit(false);
+            }
+            this.isOpen = false;
+        }
+    }
+
+    /** Opens the popover. */
+    open(): void {
+        if (!this._overlayRef || !this._overlayRef.hasAttached()) {
+            this._overlayRef = this._overlay.create(this._getOverlayConfig());
+            this._overlayRef.attach(new TemplatePortal(this.templateRef, this.container));
+
+            this._changeDetectorReference.detectChanges();
+
+            if (!this.isOpen) {
+                this.isOpenChange.emit(true);
+            }
+            this.isOpen = true;
+
+            if (this.fillControlMode) {
+                this._listenOnResize();
+            }
+
+            this._listenOnClose();
+            this._listenOnOutClicks();
+        }
+    }
+
+    /**
+     * Method called to change position of popover,
+     * recommended to be used only when popover is opened, otherwise change position or cdkPlacement
+     */
+    applyNewPosition(positions: ConnectedPosition[]): void {
+        const refPosition = this._getPositionStrategy(positions);
+        this._listenForPositionChange(refPosition.positionChanges);
```

Comment:
Don't you have multiple listeners on `positionChange` after updating position?

Reply:
I added some refresh, when this method is called